modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentParser.java

```diff
@@ -204,6 +219,14 @@ private VersionedName findMatch(List<UserAgentSubpattern> possiblePatterns, Stri
             this.patch = patch;
             this.build = build;
         }
+
+        VersionedName(String name) {
+            this.name = name;
+            this.major = null;
+            this.minor = null;
+            this.patch = null;
+            this.build = null;
+        }
```

Comment:
I'd create a test helper method rather than add a constructor that leaves four of the five class members unset.